src/main/webapp/dev/js/main/instructorCourseEnrollPage.js

```diff
@@ -122,58 +122,55 @@ function adjustStudentsPanelView(panelHeading, panelCollapse, isDataHandsontable
-
-    const ajaxPreloadImage = panelHeading.parent().find('#ajax-preload-image');
-    const ajaxStatusMessage = panelHeading.parent().find('.ajax-status-message');
-    ajaxStatusMessage.hide(); // hide any status message from the previous state
 
-    // perform AJAX only if existing students' spreadsheet is empty
-    if (getSpreadsheetLength(dataHandsontable.getData()) === 0) {
-        getAjaxStudentList(ajaxPreloadImage)
-                .then((data) => {
-                    ajaxPreloadImage.toggle();
-                    if (data.students.length === 0) {
+function studentsPanelClickHandler(currentSpreadsheet) {
+    const panelHeading = $(currentSpreadsheet.currentTarget);
+    const panelCollapse = panelHeading.parent().children('.panel-collapse');
+    const isExistingStudentsPanel = panelCollapse.find('#enrollSpreadsheet').length === 1;
+
+    if (isExistingStudentsPanel) { // Existing students' panel
+        const ajaxPreloadImage = panelHeading.find('#ajax-preload-image');
+        const ajaxStatusMessage = panelHeading.find('.ajax-status-message');
+        ajaxStatusMessage.hide(); // hide any status message from the previous state
+
+        // perform AJAX only if existing students' spreadsheet is empty
+        if (getSpreadsheetLength(dataHandsontable.getData()) === 0) {
+            getAjaxStudentList(ajaxPreloadImage)
+                    .then((data) => {
+                        ajaxPreloadImage.toggle();
+                        if (data.students.length === 0) {
+                            ajaxStatusMessage.toggle();
+                            ajaxStatusMessage.text('No existing students in course.');
+                        } else {
+                            loadExistingStudentsData(data.students);
+                            adjustStudentsPanelView(panelHeading, panelCollapse, true);
+                        }
+                    }).catch(() => {
+                        ajaxPreloadImage.toggle();
                         ajaxStatusMessage.toggle();
-                        ajaxStatusMessage.text('No existing students in course.');
-                    } else {
-                        loadExistingStudentsData(data.students);
-                        adjustStudentsPanelView(panelHeading, panelCollapse, true);
-                    }
-                }).catch(() => {
-                    ajaxPreloadImage.toggle();
-                    ajaxStatusMessage.toggle();
-                    ajaxStatusMessage.text('Failed to load. Click here to retry.');
-                });
-    } else {
+                        ajaxStatusMessage.text('Failed to load. Click here to retry.');
+                    });
+        } else {
+            adjustStudentsPanelView(panelHeading, panelCollapse, true);
+        }
+    } else { // New students' panel
         adjustStudentsPanelView(panelHeading, panelCollapse, true);
+        $('.enroll-students-spreadsheet-buttons').toggle();
```

Comment:
Now that `expandCollapseNewStudentsPanel` has become this simple, I think you should separate them into 2 functions for readability.

Anyway, I think these functions don't need to have names. `expandCollapseNewStudentsPanel` is not a very nice name anyway. You can just use anonymous functions when you register the event listener. That would be clear that this function is the handler for the event. E.g.

```javascript
$('#enroll-spreadsheet').on('click', (event) => {
    panelHeading = event.currentTarget;
    ...
});
```

Same for the other spreadsheet.